exec/vector/src/main/java/org/apache/drill/exec/vector/accessor/convert/StandardConversions.java

```diff
@@ -0,0 +1,238 @@
+     * is needed except for semantic reasons.
+     */
+    IMPLICIT,
+    /**
+     * Conversion is needed because there is no "natural",
+     * precision-preserving conversion. Conversions must be done on
+     * an ad-hoc basis.
+     */
+    EXPLICIT
+  }
+
+  /**
+   * Definition of a conversion including conversion type and the standard
+   * conversion class (if available.)
+   */
+  public static class ConversionDefn {
+    public final ConversionType type;
+    public final Class<? extends AbstractWriteConverter> conversionClass;
+
+    public ConversionDefn(ConversionType type) {
+      this.type = type;
+      conversionClass = null;
+    }
+
+    public ConversionDefn(Class<? extends AbstractWriteConverter> conversionClass) {
+      this.type = ConversionType.EXPLICIT;
+      this.conversionClass = conversionClass;
+    }
+  }
+
+  /**
+   * Column conversion factory for the case where a conversion class is provided.
+   */
+  public static class SimpleWriterConverterFactory implements ColumnConversionFactory {
+    private final Class<? extends AbstractWriteConverter> conversionClass;
+
+    SimpleWriterConverterFactory(Class<? extends AbstractWriteConverter> conversionClass) {
+      this.conversionClass = conversionClass;
+    }
+
+    @Override
+    public AbstractWriteConverter newWriter(ScalarWriter baseWriter) {
+      return newInstance(conversionClass, baseWriter);
+    }
+  }
+
+  public static ColumnConversionFactory factory(Class<? extends AbstractWriteConverter> converterClass) {
+    return new SimpleWriterConverterFactory(converterClass);
+  }
+
+  public static AbstractWriteConverter newInstance(
+      Class<? extends AbstractWriteConverter> conversionClass, ScalarWriter baseWriter) {
+    try {
+      final Constructor<? extends AbstractWriteConverter> ctor = conversionClass.getDeclaredConstructor(ScalarWriter.class);
+      return ctor.newInstance(baseWriter);
+    } catch (final ReflectiveOperationException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  /**
+   * Create converters for standard cases.
+   *
+   * @param inputDefn the column schema for the input column which the
+   * client code (e.g. reader) wants to produce
+   * @param outputDefn the column schema for the output vector to be produced
+   * by this operator
+   * @return a description of the conversion needed (if any), along with the
+   * standard conversion class, if available
+   */
+  public static ConversionDefn analyze(ColumnMetadata inputSchema, ColumnMetadata outputSchema) {
+    if (inputSchema.type().equals(outputSchema.type())) {
+      return new ConversionDefn(ConversionType.NONE);
+    }
+
+    switch (inputSchema.type()) {
+    case VARCHAR:
+      return new ConversionDefn(convertFromVarchar(outputSchema));
+    case TINYINT:
+      switch (outputSchema.type()) {
+      case SMALLINT:
+      case INT:
+      case BIGINT:
+      case FLOAT4:
+      case FLOAT8:
+      case DECIMAL18:
```

Comment:
Please exclude all decimal minor types except of `VARDECIMAL`, they are deprecated.